bitcoin/src/output.rs

```diff
@@ -0,0 +1,246 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! Transaction output types.
+
+use hashes::Hash;
+use secp256k1::{Secp256k1, Verification, Message, Signing};
+
+use crate::blockdata::script::{ScriptBuf, ScriptHash, WScriptHash};
+use crate::blockdata::witness::Witness;
+use crate::crypto::{taproot, ecdsa};
+use crate::crypto::sighash::{EcdsaSighashType, TapSighashType, SegwitV0Sighash, TapSighash, LegacySighash};
+use crate::key::{PubkeyHash, PublicKey, TweakedPublicKey, UntweakedPublicKey, WPubkeyHash, UntweakedKeypair, TweakedKeypair, TapTweak};
+use crate::taproot::TapNodeHash;
+
+/// A trait describing the various Bitcoin output types.
+pub trait Output {
+    /// The hashtype used when creating the sighash for this output.
+    type SighashType;
+    /// The sighash for this output.
+    type Sighash;
+    /// The required signature type to spend this output.
+    type Signature;
+
+}
+
+/// A P2TR output used as a key path spend (segwit v1).
+pub struct P2trKeySpend;
```

Comment:
In 82a8b55f599f529659958ad0ad74a9c441429688:

These should all be empty enums rather than empty structs because the user is not intended to actually have one of these.